python/tests/operations/reduce.py

```diff
@@ -0,0 +1,298 @@
+import os
+import unittest
+
+import equistore.io
+import numpy as np
+import equistore.operations as fn
+from equistore import Labels, TensorBlock, TensorMap
+
+
+DATA_ROOT = os.path.join(os.path.dirname(__file__), "..", "data")
+
+
+class TestReduction_sample(unittest.TestCase):
+    def test_reduction_block(self):
+        tensor_se = equistore.io.load(
+            os.path.join(DATA_ROOT, "qm7-spherical-expansion.npz"),
+            use_numpy=True,
+        )
+        tensor_ps = equistore.io.load(
+            os.path.join(DATA_ROOT, "qm7-power-spectrum.npz"),
+            use_numpy=True,
+        )
+        bl1 = tensor_ps[0]
+
+        reduce_tensor_se = fn.reduce(tensor_se, sample_name=["structure"])
+        reduce_tensor_ps = fn.reduce(tensor_ps, sample_name=["structure"])
+
+        self.assertTrue(
+            np.all(
+                np.sum(bl1.values[:4], axis=0) == reduce_tensor_ps.block(0).values[0]
+            )
+        )
+        self.assertTrue(
+            np.all(
+                np.sum(bl1.values[4:10], axis=0) == reduce_tensor_ps.block(0).values[1]
+            )
+        )
+        self.assertTrue(
+            np.all(
+                np.sum(bl1.values[10:14], axis=0) == reduce_tensor_ps.block(0).values[2]
+            )
+        )
+        self.assertTrue(
+            np.all(
+                np.sum(bl1.values[14:16], axis=0) == reduce_tensor_ps.block(0).values[3]
+            )
+        )
+        self.assertTrue(
+            np.all(
+                np.sum(bl1.values[16:22], axis=0) == reduce_tensor_ps.block(0).values[4]
+            )
+        )
+        self.assertTrue(
+            np.all(
+                np.sum(bl1.values[22:26], axis=0) == reduce_tensor_ps.block(0).values[5]
+            )
+        )
+        self.assertTrue(
+            np.all(
+                np.sum(bl1.values[26:32], axis=0) == reduce_tensor_ps.block(0).values[6]
+            )
+        )
+        self.assertTrue(
+            np.all(
+                np.sum(bl1.values[32:38], axis=0) == reduce_tensor_ps.block(0).values[7]
+            )
+        )
```

Comment:
I would not test everything one by one, it makes the test harder to read and modify later if we need to. How about testing the first new sample and one in the middle only?

Reply:
i left few of them at the beginning, in the middle and at the end